### ISSUE TITLE
Tensorflow library loading order causes segmentation fault 

### ISSUE THREAD

**Issue by shanthoosh** [tensorflow/tensorflow#22307]
My application depends on both RocksDB and tensorflow. If I load rocksDB shared library before loading the tensorflow model in my application then it results in SIGSEGV. However, if the loading order is reversed then there're no exceptions(and it works fine as expected). 

Here're the external dependencies of my application:
Tensorflow  version: 1.6
RocksDB version: 5.7.3
Java version: JDK-8
OS platform and distribution: RedHat-7(RHEL-7.2)

The following code uses both RocksDB and tensorflow results in segmentation fault:

```java
public class LibLoad{
    public static String absPath(String relativePath) {
	return System.getProperty("user.dir") + "/" + relativePath;
    }
    public static void main(String[] args) throws InterruptedException {
      System.load(LibLoad.absPath("libs/librocksdbjni6775043347215777920.so"));
      System.load(LibLoad.absPath("libs/tensorflow_native_libraries-1534022257987-0/libtensorflow_framework.so"));
    }
}
```

Actual exception:
```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000000000000, pid=3773, tid=140566680151808
#
# JRE version: Java(TM) SE Runtime Environment (8.0_05-b13) (build 1.8.0_05-b13)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.5-b02 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  0x0000000000000000
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try “ulimit -c unlimited” before starting Java again
#
# An error report file with more information is saved as:
# /export/content/lid/apps/careers-banzai-jobs-embedding-samza/dev-i001/hs_err_pid3773.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
  
6  <signal handler called>
#7  0x0000000000000000 in ?? ()
#8  0x00007f17b8dbfbb0 in pthread_once () from /lib64/libpthread.so.0
#9  0x00007f16dd60545a in void std::call_once<void (&)()>(std::once_flag&, void (&)()) ()
   from libtensorflow_framework.so
#10 0x00007f16dd60549e in tensorflow::port::TestCPUFeature(tensorflow::port::CPUFeature) ()
 
```

Similar issues had been reported previously in the community and the suggestion solution was to load tensorflow shared library before the RocksDB shared library. 

1. Just curious about the actual root cause of this segmentation fault. Can someone help me on how to get to the bottom of it.
2. What is the recommended long-term solution for this problem? Does the same codeflow that caused segmentation fault when loading the tensorflow model get triggered in other scenarios as well?

Any help for solving this problem is greatly appreciated.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by asimshankar** [comment]
CC @m3bm3b in case he has ideas. From a cursory look at the stacktrace it seems `pthread_once` is somehow leading to an invalid program address?

**Comment by m3bm3b** [comment]
I agree that debug symbols could help, if you can get a symbolic stack trace from the debugger.

pthread_once() is fairly straightforward.
I'm slightly suspicious that the libstdc++ implementation of call_once() 
is rather complicated (e.g., it uses TLS).   
I'd be tempted to try replacing the use of call_once() 
with a direct call to pthread_once() to see whether the problem is still present.

**Comment by wt-huang** [comment]
@shanthoosh Using symbolic stack trace would work, closing this and feel free to open a new issue if any other errors.

**Comment by vsooda** [comment]
you can try to upgrade rocksDB to solve this issue. it works in my situation.
